### PR TITLE
Closes #2110 Add Upgrade Status Report Step to Probo.

### LINKED ISSUE #2110: Add upgrade status check to ProboCI
## Motivation
Now that upgrade_status is included in the az-quickstart-dev package, we can use it in our automated testing to see upgrade status reports.
https://github.com/az-digital/az-quickstart-dev/commit/70f72152f07b872b03e8988decc4b11aab995160

### PR DESCRIPTION
**Pull request by trackleft**
## Description
See #2110 

On Probo
Enables upgrade status
Generates a report

## Related issues
Closes #2110

## How to test
Look in probo for the generated report.

## Types of changes
CI

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
See https://b7536a67-a099-4ffa-bcb7-8b3695e79c04.probo.build/user?destination=admin/reports/upgrade-status

File: .probo.yaml [hunk 1]
```diff
@@ -90,6 +90,13 @@ steps:
       - curl -sS -o /dev/null http://localhost/pages/annual-events
       # Check for PHP watchdog entries
       - if [ -z "$($SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -n watchdog:show --count=100 --type=php)" ]; then echo "No PHP notices/errors logged."; else echo "PHP notices/errors found in log." >&2 && exit 1; fi
+  - name: Generate upgrade status report
+    plugin: Script
+    script:
+      # Enable upgrade_status module
+      - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -y -n pm:install upgrade_status
+      # Enable upgrade_status module
```
Comment (trackleft) [inline .probo.yaml:98]:
```suggestion
      # Run report
```

File: .probo.yaml [hunk 2]
```diff
@@ -90,6 +90,13 @@ steps:
...
+      - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush upgrade_status:analyze --all
```
Comment (joeparsons) [inline .probo.yaml:99]:
```suggestion
      - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush upgrade_status:analyze az_quickstart
```

**Comment by trackleft** [comment]
Building the report now only takes 20 seconds (limited to az_quickstart)

**Comment by joeparsons** [comment]
I'm wondering if we should add steps to install our optional Quickstart modules to get rid of some of those warnings and also possibly wrap the `drush upgrade_status:analyze` step in an `if` statement similar to this one to actually cause any output to show up as a failure in ProboCI?
https://github.com/az-digital/az_quickstart/blob/main/.probo.yaml#L90

**Comment by trackleft** [comment]
Close in favor of https://github.com/az-digital/az_quickstart/pull/2135

**Pull request closed without merging**